### PR TITLE
Task 4

### PR DESCRIPTION
**Pull request by UlybinSergej**

### CODE REVIEW

File: src/task4/task1/Fruit.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    public static int getRandomWeight() {
+        Random random = new Random();
+        return random.nextInt(15) + 1;
+    }
+
+    @Override
+    public String toString() {
+        return "Fruit{" +
+                "weight=" + weight +
+                '}';
```
Comment (YanBelikov) [inline src/task4/task1/Fruit.java:21]: You can merge it in one line

File: src/task4/task2/Apple.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        Random random = new Random();
+        return new Apple(colorArray[random.nextInt(3)], Fruit.getRandomWeight());
+    }
+
+    @Override
+    public String toString() {
+        return "Apple: " +
+                "color= " + color + ", weight= " + super.weight;
+    }
+}
```
Comment (YanBelikov) [inline src/task4/task2/Apple.java:24]: You already add the same class.
Detele this

File: src/task4/task2/Fruit.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        return random.nextInt(15) + 1;
+    }
+
+    @Override
+    public String toString() {
+        return "Fruit{" +
+                "weight=" + weight +
+                '}';
+    }
+}
```
Comment (YanBelikov) [inline src/task4/task2/Fruit.java:23]: the same

File: src/task4/task2/Orange.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        Random random = new Random();
+        return new Orange(deepSkinArray[random.nextInt(21)], Fruit.getRandomWeight());
+    }
+
+    @Override
+    public String toString() {
+        return "Orange: " +
+                "deepSkin = " + deepSkin + ", weight = " + super.weight;
+    }
+}
```
Comment (YanBelikov) [inline src/task4/task2/Orange.java:30]: the same

File: src/task4/task2/Pineapple.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        Random random = new Random();
+        return new Pineapple(heightTailArray[random.nextInt(16)], Fruit.getRandomWeight());
+    }
+
+    @Override
+    public String toString() {
+        return "Pineapple: " +
+                "heightTail= " + heightTail + ", weight= " + super.weight;
+    }
+}
```
Comment (YanBelikov) [inline src/task4/task2/Pineapple.java:30]: the same

File: src/task4/task2/RedApplesFactory.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package task4.task2;
+
+import java.util.Random;
+
+public class RedApplesFactory extends AbstractFabric {
+    @Override
+    public Fruit makeFruit() {
+        Random random = new Random();
```
Comment (YanBelikov) [inline src/task4/task2/RedApplesFactory.java:8]: Why are use it here?

File: src/task4/task2/FruitMain.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        Fruit[] fruits = new Fruit[20];
+
+        for (int i = 0; i < 10; i++) {
+            for (int j = 0; j < 2; j++) {
+                fruits[i * 2 + j] = factory[i].makeFruit();
+                System.out.println(fruits[i * 2 + j].toString());
+            }
+        }
+    }
+}
```
Comment (YanBelikov) [inline src/task4/task2/FruitMain.java:30]: the same

File: src/task4/task2/AbstractFabric.java [hunk 8]
```diff
@@ -0,0 +1,5 @@
+package task4.task2;
+
+public abstract class AbstractFabric {
+    abstract public Fruit makeFruit();
+}
```
Comment (YanBelikov) [inline src/task4/task2/AbstractFabric.java:5]: delete this

File: src/task4/task2/GreenApplesFactory.java [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+
+import java.util.Random;
+
+public class GreenApplesFactory extends AbstractFabric {
+    @Override
+    public Fruit makeFruit() {
+        Random random = new Random();
+        return new Apple("Green", Fruit.getRandomWeight());
+    }
+}
```
Comment (YanBelikov) [inline src/task4/task2/GreenApplesFactory.java:11]: delete this

File: src/task4/task2/LongPineapplesFactory.java [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+
+import java.util.Random;
+
+public class LongPineapplesFactory extends AbstractFabric {
+    @Override
+    public Fruit makeFruit() {
+        Random random = new Random();
+        return new Pineapple(random.nextInt(6) + 10, Fruit.getRandomWeight());
+    }
+}
```
Comment (YanBelikov) [inline src/task4/task2/LongPineapplesFactory.java:11]: delete this

File: src/task4/task2/RedApplesFactory.java (hunk 6 above)
Comment (YanBelikov) [inline src/task4/task2/RedApplesFactory.java:8]: delete this

File: src/task4/task2/ThinOrangeFactory.java [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+
+import java.util.Random;
+
+public class ThinOrangeFactory extends AbstractFabric {
+    @Override
+    public Fruit makeFruit() {
+        Random random = new Random();
+        return new Orange(random.nextInt(5) + 1, Fruit.getRandomWeight());
+    }
+}
```
Comment (YanBelikov) [inline src/task4/task2/ThinOrangeFactory.java:11]: delete this

File: src/task4/task1/Fruit.java (hunk 1 above)
Comment (UlybinSergej) [reply src/task4/task1/Fruit.java:21]: Done

File: src/task4/task2/AbstractFabric.java (hunk 8 above)
Comment (UlybinSergej) [reply src/task4/task2/AbstractFabric.java:5]: Delete class, or delete method?
Text of task:
"Создать класс AbstractFabric с одним абстрактным методом makeFruit(), который возвращает фрукт."

**Review by YanBelikov — APPROVED** [review]

**Pull request merged**